pageserver/src/walredo.rs

```diff
@@ -187,10 +188,27 @@ impl PostgresRedoManager {
         PostgresRedoManager {
             tenant_id,
             conf,
+            last_redo_at: std::sync::Mutex::default(),
             redo_process: RwLock::new(None),
         }
     }
 
+    /// This type doesn't have its own background task to check for idleness: we
+    /// rely on our owner calling this function periodically in its own housekeeping
+    /// loops.
+    pub(crate) fn maybe_quiesce(&self, idle_timeout: Duration) {
+        if let Ok(g) = self.last_redo_at.try_lock() {
+            if let Some(last_redo_at) = *g {
+                if last_redo_at.elapsed() < idle_timeout {
+                    return;
+                }
+            }
+        }
+
+        let mut guard = self.redo_process.write().unwrap();
+        *guard = None;
```

Comment:
these two lines need to be after line 205
if not , this would be called even if the try_lock returned WouldBlock

Reply:
Good point, that would had been nice one to debug :) Fixed in 4d0b09161